### PR TITLE
added bunch of helper modules for future expansion

### PR DESCRIPTION
**Pull request by hoonsubin**
made the code flexible for future expansion. I've separated the component logic from the helper logic so all we have to do is just implement the lock/lockdrop helper modules with their respective libraries.
You can change the options to be something else as well.

TODOs:
- make better visuals
- make warning page for incorrect inputs
- **implement the actual contract communication logic**

### CODE REVIEW

File: app/src/pages/LockdropPage.tsx [hunk 1]
```diff
@@ -1,99 +1,206 @@
...
+	onChoose: Function;
+};
 
-const contractAddress = '0xFEC6F679e32D45E22736aD09dFdF6E3368704e31';
-const durations: string[] = ['3 Months', '6 Months', '12 Months', 'Forever my dear!'];
-const txType: string[] = ['Metamask', 'Plasm CLI', 'MYCrypto'];
-const tokenTypes: string[] = ['ETH', 'BTC', 'DOT'];
+const ethContractAddr = '0xFEC6F679e32D45E22736aD09dFdF6E3368704e31';
 
+const durations: OptionItem[] = [
```
Comment (satellitex) [inline app/src/pages/LockdropPage.tsx]: I think it should be adding `rate`.
```
{ label: '30 Days', value: 30, rate: 24 },
{ label: '100 Days', value: 100, rate: 100 },
{ label: '300 Days', value: 300, rate: 360 },
{ label: '1000 Days', value: 1000, rate: 1600 },
```
And, this rate is viewed to users.

**Comment by akru** [comment]
I'm not sure that it should be all-in-one dapp for ETH/BTC/EOS/etc. What's about to focus on ETH right now and make it minimalistic but workable.
I guess also should be page to release already locked tokens.

**Comment by akru** [comment]
@hoonsubin could your resolve merge conflicts please.

**Comment by hoonsubin** [comment]
@akru okay, will do asap

**Comment by akru** [comment]
@hoonsubin Thanks, did you finish this PR?

**Comment by hoonsubin** [comment]
bandicam 2020-02-01 11-57-32-767
Finished implementing MetaMask connection via Drizzle.

Todo:

- display token increase rate
- communicate with smart contract
- add loading screen
- authenticate user input
- test

**Comment by hoonsubin** [comment]
@satellitex @akru 
Error in app/src/helpers/lockdrop/EthereumLockdrop.ts line 10.
 
`Type '{ "contractName": string; "abi": ({ "inputs": { "internalType": string; "name": string; "type": string; }[]; "payable": boolean; "stateMutability": string; "type": string; } | { "anonymous": boolean; "inputs": { "indexed": boolean; "internalType": string; "name": string; "type": string; }[]; "name": string; "type": ...' is not assignable to type 'IContract'.
  Types of property 'abi' are incompatible.
    Type '({ "inputs": { "internalType": string; "name": string; "type": string; }[]; "payable": boolean; "stateMutability": string; "type": string; } | { "anonymous": boolean; "inputs": { "indexed": boolean; "internalType": string; "name": string; "type": string; }[]; "name": string; "type": string; } | { ...; } | { ...; })[]' is not assignable to type 'ABI[]'.
      Type '{ "inputs": { "internalType": string; "name": string; "type": string; }[]; "payable": boolean; "stateMutability": string; "type": string; } | { "anonymous": boolean; "inputs": { "indexed": boolean; "internalType": string; "name": string; "type": string; }[]; "name": string; "type": string; } | { ...; } | { ...; }' is not assignable to type 'ABI'.
        Type '{ "inputs": { "internalType": string; "name": string; "type": string; }[]; "payable": boolean; "stateMutability": string; "type": string; }' is missing the following properties from type 'ABI': constant, name, outputs`

every time when I try to add a smart contract abi to the array

File: app/src/pages/EthLockdropPage.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+const durations: OptionItem[] = [
+	{ label: '30 Days', value: 30 },
+	{ label: '100 Days', value: 100 },
+	{ label: '300 Days', value: 300 },
+	{ label: '1000 Days', value: 1000 }
+];
+
+const txTypes: OptionItem[] = [{ label: 'Metamask', value: 'metamask' }];
+
+const rates = [
```
Comment (satellitex) [inline app/src/pages/EthLockdropPage.tsx]: Currently, These rates are hidden from users.
We should explicitly show these rates to users, I think.

Comment (hoonsubin) [reply app/src/pages/EthLockdropPage.tsx]: resolved

**Pull request merged**